phoenix-core-client/src/main/java/org/apache/phoenix/expression/CoerceExpression.java

```diff
@@ -145,11 +147,30 @@ public void write(DataOutput output) throws IOException {
 
     @Override
     public boolean evaluate(Tuple tuple, ImmutableBytesWritable ptr) {
-        if (getChild().evaluate(tuple, ptr)) {
-            getDataType().coerceBytes(ptr, null, getChild().getDataType(),
-                    getChild().getMaxLength(), null, getChild().getSortOrder(), 
-                    maxLength, null, getSortOrder(), rowKeyOrderOptimizable);
-            return true;
+        if (getChild() instanceof LiteralExpression
+            && getChild().getDataType() == PVarbinary.INSTANCE
```

Comment:
what is this check fo?

Reply:
I will write more comments, filling up design doc section on this as well.